blazor/datagrid/column-headers.md

```diff
@@ -0,0 +1,846 @@
+                    Orders.Add(new OrderData(10248, "RICSU", new DateTime(1996, 07, 08),55.45));
+                    Orders.Add(new OrderData(10248, "VINET", new DateTime(1996, 07, 08),11.94));                                                                                    
+                    code += 5;
+                }
+            }
+            return Orders;
+        }
+        public int? OrderID { get; set; }
+        public string CustomerID { get; set; }
+        public DateTime? OrderDate { get; set; }
+        public double? Freight { get; set; }
+    }
+{% endhighlight %}
+{% endtabs %}
+
+{% previewsample "https://blazorplayground.syncfusion.com/embed/LNrUMtACJAAfwAum?appbar=false&editor=false&result=true&errorlist=false&theme=bootstrap5" %}
+
+
+>* The `HeaderText` property is optional, and if it is not defined, then the corresponding column's field value is set as header text for that column.  
+>* You can also use the [HeaderTemplate](https://help.syncfusion.com/cr/blazor/Syncfusion.Blazor.Grids.GridColumn.html#Syncfusion_Blazor_Grids_GridColumn_HeaderTemplate) of the `GridColumn` component to apply custom HTML content to the header cell.
+
+## Header template
+
+The [HeaderTemplate](https://help.syncfusion.com/cr/blazor/Syncfusion.Blazor.Grids.GridColumn.html#Syncfusion_Blazor_Grids_GridColumn_HeaderTemplate) of the  [GridColumn](https://help.syncfusion.com/cr/blazor/Syncfusion.Blazor.Grids.GridColumn.html) component used to customize the header element of a Grid column. With this property, you can render custom HTML elements or Blazor components to the header element. This feature allows you to add more functionality to the header, such as sorting or filtering.
+
+To know about **Header Template** in Blazor DataGrid Component, you can check this video.
+
+{% youtube
+"youtube:https://www.youtube.com/watch?v=9YF9HnFY5Ew"%}
+
+In this demo, the custom element is rendered for both **EmployeeID** and **OrderDate** column headers.
+
+{% tabs %}
+{% highlight razor tabtitle="Index.razor" %}
+@using Syncfusion.Blazor.Grids
+@using Syncfusion.Blazor.Buttons
+@using BlazorApp1.Data
+
+<SfGrid @ref="Grid" DataSource="@Orders" AllowPaging="true" Height="315">
+    <GridColumns>
+        <GridColumn Field=@nameof(OrderData.OrderID) TextAlign="TextAlign.Right" Width="120"></GridColumn>
+        <GridColumn Field=@nameof(OrderData.CustomerID) Width="130">
+            <HeaderTemplate>
+                <div>
+                    <span class="e-icon-userlogin e-icons employee"></span> Customer ID
+                </div>
+            </HeaderTemplate>
+        </GridColumn>
+        <GridColumn Field=@nameof(OrderData.Freight) HeaderText="Freight" Format="C2" Width="150"></GridColumn>
+        <GridColumn Field=@nameof(OrderData.OrderDate) Format="d" TextAlign="TextAlign.Right" Width="140">
+            <HeaderTemplate>
+                <div >
+                    <SfSwitch ValueChange="Change" TChecked="bool"></SfSwitch>@HeaderValue
```

Comment:
alignment is not proper

Reply:
Due to length of the column , it shows like this.